ibm_i/datadog_checks/ibm_i/check.py

```diff
@@ -78,17 +69,125 @@ def check(self, _):
                 hostname=self._query_manager.hostname,
             )
 
-    def execute_query(self, query):
-        # https://github.com/mkleehammer/pyodbc/wiki/Connection#execute
-        with closing(self.connection.execute(query)) as cursor:
+    def cancel(self):
+        # When the check gets cancelled, clean up the connection subprocess.
+        self._delete_connection_subprocess()
+
+    def handle_query_error(self, error):
+        self._current_errors += 1
+        return error
+
+    def _create_connection_subprocess(self):
```

Comment:
Let's access like an attribute:

```suggestion

    @property
    def connection_subprocess(self):
        if self._subprocess is None:
            self._create_connection_subprocess()

        return self._subprocess

    def _create_connection_subprocess(self):
```